app/src/main/java/org/p2p/wallet/rpc/RpcModule.kt

```diff
@@ -1,50 +1,80 @@
+import org.p2p.wallet.rpc.repository.balance.RpcBalanceRepository
+import org.p2p.wallet.rpc.repository.blockhash.RpcBlockHashRemoteRepository
+import org.p2p.wallet.rpc.repository.blockhash.RpcBlockHashRepository
+import org.p2p.wallet.rpc.repository.signature.RpcSignatureRemoteRepository
+import org.p2p.wallet.rpc.repository.signature.RpcSignatureRepository
+import org.p2p.wallet.rpc.repository.history.RpcHistoryRemoteRepository
+import org.p2p.wallet.rpc.repository.history.RpcHistoryRepository
+import retrofit2.Retrofit
 
 object RpcModule : InjectionModule {
 
+    const val RPC_RETROFIT_QUALIFIER = "RPC_RETROFIT_QUALIFIER"
+
     override fun create() = module {
-        single {
-            val serverErrorInterceptor = ServerErrorInterceptor(get())
-            val serum = getRetrofit(Environment.SOLANA.endpoint, interceptor = serverErrorInterceptor)
-            val serumRpcApi = serum.create(RpcApi::class.java)
-
-            val mainnet = getRetrofit(Environment.MAINNET.endpoint, interceptor = serverErrorInterceptor)
-            val mainnetRpcApi = mainnet.create(RpcApi::class.java)
-
-            /* This string is in gitignore and it's null when CI/CD runs some actions */
-            val rpcPoolApiKey = BuildConfig.rpcPoolApiKey
-            val baseUrl = if (rpcPoolApiKey.isNotBlank()) {
-                "${Environment.RPC_POOL.endpoint}$rpcPoolApiKey/"
-            } else {
-                Environment.RPC_POOL.endpoint
-            }
-            val rpcpool = getRetrofit(baseUrl, interceptor = serverErrorInterceptor)
-            val rpcpoolRpcApi = rpcpool.create(RpcApi::class.java)
-
-            val testnet = getRetrofit(Environment.DEVNET.endpoint, interceptor = serverErrorInterceptor)
-            val testnetRpcApi = testnet.create(RpcApi::class.java)
-
-            RpcRemoteRepository(serumRpcApi, mainnetRpcApi, rpcpoolRpcApi, testnetRpcApi, get())
-        } bind RpcRepository::class
-
-        factory { TokenInteractor(get(), get(), get()) }
-        factory { TransactionInteractor(get(), get(), get()) }
-
-        single { RpcAmountRemoteRepository(get()) } bind RpcAmountRepository::class
+        factory {
+            val api = get<Retrofit>(named(RPC_RETROFIT_QUALIFIER)).create(RpcAccountApi::class.java)
+            RpcAccountRemoteRepository(api)
+        } bind RpcAccountRepository::class
+
+        factory {
+            val api = get<Retrofit>(named(RPC_RETROFIT_QUALIFIER)).create(RpcAmountApi::class.java)
+            RpcAmountRemoteRepository(api)
+        } bind RpcAmountRepository::class
+
+        single { RpcAmountInMemoryRepository() } bind RpcAmountLocalRepository::class
+
+        factory { RpcAmountInteractor(get(), get()) }
+
+        factory {
+            val api = get<Retrofit>(named(RPC_RETROFIT_QUALIFIER)).create(RpcBalanceApi::class.java)
+            RpcBalanceRemoteRepository(api)
+        } bind RpcBalanceRepository::class
+
+        factory {
```

Comment:
im not sure that the Repository entity should be injected using a `factory`. I think it's more of a `single`.